src/datarobotx/idp/common/credentials_hooks.py

```diff
@@ -25,27 +26,46 @@
 class CredentialsHooks:
     """Include credentials in the catalog so they can be used by nodes."""
 
+    @property
+    def _logger(self):
+        return logging.getLogger(__name__)
+
     @hook_impl
     def after_catalog_created(
         self,
         catalog: DataCatalog,
-        conf_catalog: Dict[str, Any],
         conf_creds: Dict[str, Any],
-        save_version: str,
-        load_versions: Dict[str, str],
     ) -> None:
         """Validate credentials and add them to the catalog for node consumption."""
-        self.validate_credentials(conf_creds)
+        self._validate_or_replace_credentials(conf_creds)
 
-        creds = get_feed_dict(conf_creds, copy_dict_as=None, key_prefix="params:credentials.")
+        creds = get_feed_dict(
+            conf_creds, copy_dict_as=None, key_prefix="params:credentials."
+        )
         catalog.add_feed_dict(creds)
 
-    @staticmethod
-    def validate_credentials(conf_creds: Dict[str, Any]) -> None:
+    def _validate_or_replace_credentials(self, conf_creds: Dict[str, Any]) -> None:
         """Confirm credentials appear to be valid and functional."""
         import datarobot as dr
 
         dr_creds = conf_creds["datarobot"]
+        try:
+            dr.Client(token=dr_creds["api_token"], endpoint=dr_creds["endpoint"])  # type: ignore
+        except ValueError:
+            self._logger.warning(
+                "Could not authenticate with DataRobot. "
+                "Kedro will run dr.Client() for authentication."
+            )
+            client = dr.Client()
+            conf_creds["datarobot"]["api_token"] = client.token
+            conf_creds["datarobot"]["endpoint"] = client.endpoint
+            prediction_server_id = dr.PredictionServer.list()[0].id
 
-        client = dr.Client(token=dr_creds["api_token"], endpoint=dr_creds["endpoint"])  # type: ignore
```

Comment:
don't we still need the test that the client works somewhere here?

Reply:
I learned that `/version` is actually called when you run `dr.Client()` so no, the test is done implicitly by instantiating the client https://datarobot.slack.com/archives/C0E80BJS0/p1724853667451609